src/schema/query/application.query.ts

```diff
@@ -33,13 +39,17 @@ export default {
         .where({ _id: args.id })
         .getFilter();
       const application = await Application.findOne(filters);
-      if (application && args.asRole) {
-        const pages: Page[] = await Page.aggregate([
+      if (application && (args.isFrontOfficeModule || args.asRole)) {
+        const aggregateOptions: mongoose.PipelineStage[] = [
           {
             $match: {
-              'permissions.canSee': {
-                $elemMatch: { $eq: new mongoose.Types.ObjectId(args.asRole) },
-              },
+              ...(args.asRole && {
+                'permissions.canSee': {
+                  $elemMatch: { $eq: new mongoose.Types.ObjectId(args.asRole) },
+                },
+              }),
+              /** Include only visible pages for front office module */
+              ...(args.isFrontOfficeModule && { visible: true }),
```

Comment:
@unai-reliefapp 
not an option
the point is not the remove the pages from the query, but to hide them, so I don't think there is any back-end work to do

Reply:
yes, because some logic is built around them
the visible attribute is just there to tell if yes or not a page is visible

Reply:
So just html hide, no filter of pages in the ts files or anything similar @AntoineRelief

Reply:
yes
and check in the code, I'm pretty sure we already kind of use the visible property as it's working for web element integration